logs/logs.go

```diff
@@ -56,13 +56,15 @@ type LogAgent struct {
 	backends    map[string]LogBackend
 	destNames   map[LogDest]string
 	collections []LogCollection
+	retentionAlreadySet map[string]bool
 }
 
 func NewLogAgent(c *config.Config) *LogAgent {
 	return &LogAgent{
 		Config:    c,
 		backends:  make(map[string]LogBackend),
 		destNames: make(map[LogDest]string),
+		retentionAlreadySet: make(map[string]bool),
```

Comment:
I'm not fond of use cache with map since the more entries we add, the [more bytes](https://stackoverflow.com/questions/15313105/memory-overhead-of-maps-in-go) we have to consume unless we have any mechanism to delete the entries the map after certain amount of time. Therefore, I'm still standby create the resources beforehand. (correct me if I am wrong)

Reply:
@khanhntd I think that it would be a concern if we're storing a lot of data but we are only storing the log group name and a boolean. In a sample `map[string]int` of 100 initial entries the memory usage is [4kb](https://stackoverflow.com/questions/46278003/how-much-memory-do-golang-maps-reserve#:~:text=A%20map%20of%20type%20map,7%20if%20not%20specified%20explicitly.). We should be okay in terms of memory usage.

Reply:
If that's the case then this is fine. Maybe its only my design perspective but this would introduce special code for only this use case (in this case, the `retentionAlreadySet`) but yeah, if everyone is fine with the current approach then I'm good with it.